cdap-watchdog/src/main/java/co/cask/cdap/logging/logbuffer/LogBufferWriter.java

```diff
@@ -61,19 +65,20 @@
   private long currFileId;
   private long writtenBytes;
 
-  public LogBufferWriter(String logBufferBaseDir, long maxFileSize) throws IOException {
+  public LogBufferWriter(String logBufferBaseDir, long maxFileSize, Runnable cleaner) throws IOException {
     File baseDir = new File(logBufferBaseDir);
     // make sure base dir already exists, if not create it.
     Files.createDirectories(baseDir.toPath());
     this.locationFactory = new LocalLocationFactory(baseDir);
     // max file size after which rotation should happen.
     this.maxFileSizeInBytes = maxFileSize;
+    this.cleaner = cleaner;
+    this.executorService = Executors.newSingleThreadExecutor();
```

Comment:
This should be daemon thread executor so that it does not prevent shutdown. Also would be good to name this thread.

Reply:
Also a good idea to shutdown the executor in the close method.